### PR TITLE
add envrionment depends on Tinymce Module

### PR DESCRIPTION
**Pull request by Soldras**
Allow Tinymce to save and get images or documents from different environment.

By default all files are save in web/media/upload folder, now you can set THELIA_ENV var in your vhost to save files in web/media/THELIA_ENV/upload.

### CODE REVIEW

File: local/modules/Tinymce/Tinymce.php [hunk 1]
```diff
@@ -41,6 +43,16 @@ public function postActivation(ConnectionInterface $con = null)
     {
         $fileSystem = new Filesystem();
 
+        //Check for environment
+        if ($env = $this->getContainer()->getParameter('kernel.environment')) {       
+            //Check for backward compatibility
+            if(strcmp($env,"prod") !== 0){
```
Comment (lovenunu) [inline local/modules/Tinymce/Tinymce.php]: As we benchmarked together, This is clearly not as efficient as ```$env === "prod"``` :-1:

File: local/modules/Tinymce/Tinymce.php [hunk 2]
```diff
@@ -51,11 +63,18 @@ public function postActivation(ConnectionInterface $con = null)
...
             }
         }
-
-        // Create the media directory in the web root, if required
-        if (false === $fileSystem->exists($this->webMediaPath)) {
-            $fileSystem->mkdir($this->webMediaPath . DS . 'upload');
-            $fileSystem->mkdir($this->webMediaPath . DS . 'thumbs');
+        
+        // Create the media directory in the web root , if required
+        if(null != $this->webMediaEnvPath){
```
Comment (lovenunu) [inline local/modules/Tinymce/Tinymce.php]: !==

File: local/modules/Tinymce/Tinymce.php [hunk 3]
```diff
@@ -51,11 +63,18 @@ public function postActivation(ConnectionInterface $con = null)
...
+            if(false==  $fileSystem->exists($this->webMediaEnvPath)){
```
Comment (lovenunu) [inline local/modules/Tinymce/Tinymce.php]: false ===

File: local/modules/Tinymce/Tinymce.php [hunk 4]
```diff
@@ -51,11 +63,18 @@ public function postActivation(ConnectionInterface $con = null)
...
+                $fileSystem->mkdir($this->webMediaEnvPath . DS . 'upload');
+                $fileSystem->mkdir($this->webMediaEnvPath . DS . 'thumbs');
+            }
+        }else {
```
Comment (lovenunu) [inline local/modules/Tinymce/Tinymce.php]: } else {

File: local/modules/Tinymce/Resources/js/tinymce/filemanager/config/config.php [hunk 5]
```diff
@@ -69,47 +69,63 @@
 // Argh, url_site is not defined ?!
 if (empty($base_url)) {
     // A we did not used the router to access this dialog, we cannot use the URL class. Use the good old method.
-    $base_url = $request->getSchemeAndHttpHost() . preg_replace('!/tinymce/filemanager/dialog.php.*$!', '', $_SERVER['REQUEST_URI']);
+    $base_url = $request->getSchemeAndHttpHost().preg_replace('!/tinymce/filemanager/dialog.php.*$!', '',
+            $_SERVER['REQUEST_URI']);
```
Comment (lunika) [inline local/modules/Tinymce/Resources/js/tinymce/filemanager/config/config.php]: why only this parameter on a new line ? Put all parameters on a new line or none.

File: local/modules/Tinymce/Tinymce.php [hunk 6]
```diff
@@ -41,6 +43,16 @@ public function postActivation(ConnectionInterface $con = null)
     {
         $fileSystem = new Filesystem();
 
+        //Check for environment
+        if ($env = $this->getContainer()->getParameter('kernel.environment')) {
+            //Check for backward compatibility
+            if ($env !== "prod") {
```
Comment (lunika) [inline local/modules/Tinymce/Tinymce.php]: this particular case doesn't cover default Thelia's behaviour. I can be in development mode and the folder ```media/dev``` will be created and never used. Moreover, the ```_dev``` part is never removed in the config file, so were files will be copied ?

**Comment by lunika** [comment]
Once activated, if I change my environment value, it's not possible to upload a file anymore. I have this error message : 

```
There is an error! The upload folder there isn't. Check your config.php file. 
```

File: local/modules/Tinymce/Tinymce.php (hunk 6 above)
Comment (Soldras) [inline local/modules/Tinymce/Tinymce.php]: In the config.php we never add _dev to the environment so for exemple:

you have an environment TEST_THELIA, in dev mode you TEST_THELIA_dev but tinymce must not be dependent to dev / prod so it only use TEST_THELIA. So files will be copied in ```media/TEST_THELIA```

**Comment by Soldras** [comment]
If you add an environment you have to activate the module for this environment ... It's the goal

**Comment by lunika** [comment]
Well, in your particular case, not if you have a normal use of Thelia.

**Comment by lunika** [comment]
Once a module activated, this module module is available for all the environments. In your case not because you have one database for each environment.

File: local/modules/Tinymce/Tinymce.php (hunk 6 above)
Comment (lunika) [inline local/modules/Tinymce/Tinymce.php]: ok I understand

Comment (lunika) [inline local/modules/Tinymce/Tinymce.php]: but the dev folder is still created and never used :-)

**Comment by bilhackmac** [comment]
It's useless to have multiples environnements with the same database :expressionless:

**Pull request merged**